### ISSUE TITLE
Unreasonable result from classify.py

### ISSUE THREAD

**Issue by zhangjunjie** [BVLC/caffe#2980]
I met a odd situation.
I use caffe and pretrained model to do classification. It a two class problem. I finetuned a model and use the model to classify images. However, the classification results of images are always [0.5, 0.5].
The following is my deploy.prototxt:
name: "VGG_ILSVRC_16_layers_Caltech"
input: "data"
input_dim: 1
input_dim: 3
input_dim: 128
input_dim: 64
layers {
  bottom: "data"
  top: "conv1_1"
  name: "conv1_1"
  type: CONVOLUTION
  convolution_param {
    num_output: 64
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv1_1"
  top: "conv1_1"
  name: "relu1_1"
  type: RELU
}
layers {
  bottom: "conv1_1"
  top: "conv1_2"
  name: "conv1_2"
  type: CONVOLUTION
  convolution_param {
    num_output: 64
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv1_2"
  top: "conv1_2"
  name: "relu1_2"
  type: RELU
}
layers {
  bottom: "conv1_2"
  top: "pool1"
  name: "pool1"
  type: POOLING
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layers {
  bottom: "pool1"
  top: "conv2_1"
  name: "conv2_1"
  type: CONVOLUTION
  convolution_param {
    num_output: 128
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv2_1"
  top: "conv2_1"
  name: "relu2_1"
  type: RELU
}
layers {
  bottom: "conv2_1"
  top: "conv2_2"
  name: "conv2_2"
  type: CONVOLUTION
  convolution_param {
    num_output: 128
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv2_2"
  top: "conv2_2"
  name: "relu2_2"
  type: RELU
}
layers {
  bottom: "conv2_2"
  top: "pool2"
  name: "pool2"
  type: POOLING
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layers {
  bottom: "pool2"
  top: "conv3_1"
  name: "conv3_1"
  type: CONVOLUTION
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv3_1"
  top: "conv3_1"
  name: "relu3_1"
  type: RELU
}
layers {
  bottom: "conv3_1"
  top: "conv3_2"
  name: "conv3_2"
  type: CONVOLUTION
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv3_2"
  top: "conv3_2"
  name: "relu3_2"
  type: RELU
}
layers {
  bottom: "conv3_2"
  top: "conv3_3"
  name: "conv3_3"
  type: CONVOLUTION
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
  }
}
layers {
  bottom: "conv3_3"
  top: "conv3_3"
  name: "relu3_3"
  type: RELU
}
layers {
  bottom: "conv3_3"
  top: "pool3"
  name: "pool3"
  type: POOLING
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layers {
  bottom: "pool3"
  top: "fc6_Caltech"
  name: "fc6_Caltech"
  type: INNER_PRODUCT
  inner_product_param {
    num_output: 512
  }
}
layers {
  bottom: "fc6_Caltech"
  top: "fc6_Caltech"
  name: "relu6"
  type: RELU
}
layers {
  bottom: "fc6_Caltech"
  top: "fc6_Caltech"
  name: "drop6"
  type: DROPOUT
  dropout_param {
    dropout_ratio: 0.5
  }
}
layers {
  bottom: "fc6_Caltech"
  top: "fc8_Caltech"
  name: "fc8_Caltech"
  type: INNER_PRODUCT
  inner_product_param {
    num_output: 2
  }
}
layers {
  bottom: "fc8_Caltech"
  top: "prob"
  name: "prob"
  type: SOFTMAX
}

Part of classifying log info:
...........................
I0826 15:08:58.878199  3266 net.cpp:120] Setting up prob
I0826 15:08:58.878219  3266 net.cpp:127] Top shape: 1 2 (2)
I0826 15:08:58.878231  3266 net.cpp:194] prob does not need backward computation.
I0826 15:08:58.878239  3266 net.cpp:194] fc8_Caltech does not need backward computation.
I0826 15:08:58.878248  3266 net.cpp:194] drop6 does not need backward computation.
I0826 15:08:58.878257  3266 net.cpp:194] relu6 does not need backward computation.
I0826 15:08:58.878264  3266 net.cpp:194] fc6_Caltech does not need backward computation.
I0826 15:08:58.878273  3266 net.cpp:194] pool3 does not need backward computation.
I0826 15:08:58.878281  3266 net.cpp:194] relu3_3 does not need backward computation.
I0826 15:08:58.878289  3266 net.cpp:194] conv3_3 does not need backward computation.
I0826 15:08:58.878298  3266 net.cpp:194] relu3_2 does not need backward computation.
I0826 15:08:58.878306  3266 net.cpp:194] conv3_2 does not need backward computation.
I0826 15:08:58.878314  3266 net.cpp:194] relu3_1 does not need backward computation.
I0826 15:08:58.878322  3266 net.cpp:194] conv3_1 does not need backward computation.
I0826 15:08:58.878331  3266 net.cpp:194] pool2 does not need backward computation.
I0826 15:08:58.878340  3266 net.cpp:194] relu2_2 does not need backward computation.
I0826 15:08:58.878350  3266 net.cpp:194] conv2_2 does not need backward computation.
I0826 15:08:58.878358  3266 net.cpp:194] relu2_1 does not need backward computation.
I0826 15:08:58.878367  3266 net.cpp:194] conv2_1 does not need backward computation.
I0826 15:08:58.878376  3266 net.cpp:194] pool1 does not need backward computation.
I0826 15:08:58.878386  3266 net.cpp:194] relu1_2 does not need backward computation.
I0826 15:08:58.878394  3266 net.cpp:194] conv1_2 does not need backward computation.
I0826 15:08:58.878403  3266 net.cpp:194] relu1_1 does not need backward computation.
I0826 15:08:58.878412  3266 net.cpp:194] conv1_1 does not need backward computation.
I0826 15:08:58.878420  3266 net.cpp:235] This network produces output prob
I0826 15:08:58.878443  3266 net.cpp:482] Collecting Learning Rate and Weight Decay.
I0826 15:08:58.878458  3266 net.cpp:247] Network initialization done.
I0826 15:08:58.878465  3266 net.cpp:248] Memory required for data: 16652304
Loading file: /home/qhu/USA10x_VGG_finetune_patches/test/cropped_neg_images/000002.png
Classifying 1 inputs.
Done in 2.74 s.
Saving results into res
zjj@zjj:~/dev/tmp/caffe/python$ 

zjj@zjj:~/dev/tmp/caffe/python$ python 
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.load('res.npy')
array([[ 0.5,  0.5]], dtype=float32)

I used bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel to classify "cat.jpg", it got the right result.

Any suggestion?

**Comment by alviur** [comment]
I had this issue changing names  of the layers in Image-net architecture and I could solve it. If anyone have and idea of the error, it would be a great help.

**Comment by longjon** [comment]
Closing as this looks like a usage issue/request for help.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
